test/puppetlabs/puppetdb/cli/import_export_roundtrip_test.clj

```diff
@@ -1,143 +1,101 @@
-            [puppetlabs.puppetdb.admin :as admin]
+            [puppetlabs.puppetdb.testutils.catalogs :as tuc]
+            [puppetlabs.puppetdb.testutils.services :as svc-utils]
+            [puppetlabs.puppetdb.testutils.cli :refer :all]
+            [puppetlabs.puppetdb.fixtures :as fixt]
             [slingshot.slingshot :refer [throw+ try+]]
-            [slingshot.test]
-            [puppetlabs.puppetdb.testutils.services :as svc-utils]))
+            [slingshot.test]))
 
 (use-fixtures :each fixt/with-test-logging-silenced)
 
-(defn block-until-queue-empty
-  "Blocks the current thread until all messages from the queue have been processed."
-  []
-  (loop [depth (svc-utils/current-queue-depth)]
-    (when (< 0 depth)
-      (Thread/sleep 10)
-      (recur (svc-utils/current-queue-depth)))))
-
 (deftest test-basic-roundtrip
-  (let [certname "foo.local"
-        facts {:certname certname
-               :environment "DEV"
-               :values {:foo "the foo"
-                        :bar "the bar"
-                        :baz "the baz"
-                        :biz {:a [3.14 2.71] :b "the b" :c [1 2 3] :d {:e nil}}}
-               :producer_timestamp (to-string (now))}
-        export-out-dir (tu/temp-dir "export-test")
-        export-out-file-path (str (.getPath export-out-dir) "/outfile.tar.gz")
-        catalog (-> (get-in wire-catalogs [6 :empty])
-                    (assoc :certname certname
-                           :producer_timestamp (now)))
-        report (-> (:basic reports)
-                   (assoc :certname certname))]
-
+  (let [export-out-file (.getPath (tu/temp-file "export-test" ".tar.gz"))]
     (svc-utils/call-with-single-quiet-pdb-instance
      (fn []
-       (let [query-fn (partial query (tk-app/get-service svc-utils/*server* :PuppetDBServer))
-             submit-command-fn (partial submit-command (tk-app/get-service svc-utils/*server* :PuppetDBCommand))]
-         (is (empty? (query-fn :nodes admin/query-api-version nil nil doall)))
+       (is (empty? (get-nodes)))
 
-         (svc-utils/sync-command-post
-           (tu/command-base-url svc-utils/*base-url*)
-           "replace catalog" 6 catalog)
+       (svc-utils/sync-command-post (svc-utils/pdb-cmd-url) "replace catalog" 6 example-catalog)
+       (svc-utils/sync-command-post (svc-utils/pdb-cmd-url) "store report" 5 example-report)
+       (svc-utils/sync-command-post (svc-utils/pdb-cmd-url) "replace facts" 4 example-facts)
 
-         (svc-utils/sync-command-post
-           (tu/command-base-url svc-utils/*base-url*)
-           "store report" 5
-           (tur/munge-example-report-for-storage report))
+       (is (= (tuc/munge-catalog example-catalog)
+              (tuc/munge-catalog (get-catalogs example-certname))))
+       (is (= (tur/munge-report example-report)
+              (tur/munge-report (get-reports example-certname))))
+       (is (= (tuf/munge-facts example-facts)
+              (tuf/munge-facts (get-factsets example-certname))))
```

Comment:
So this is what we actually had here originally before I changed the export tool to use in-process querying. I actually wanted to get rid of using the in-process querying for checking whether command submission was successful because it meant that every time the export tool changed that these tests had to change, whether or not the functionality of the export tool changed. I actually was going to make a ticket about getting more granular export/import tar testing, rather than only roundtrip tests like we have now. I was holding off though because I'll have to make those tests for using anonymization with the export tool (I.e. tests to make sure your export utils and the final tarball give you what you expect.)